### PR TITLE
Add nuspec for #3.

### PR DESCRIPTION
**Pull request by binki**
I added a quirky nuspec which generates a nupkg in the way I think it should. The script, `nugetpack.cmd`, ensures that the assembly is built and packed from the correct build configuration. The nuspec makes the assumption that you will continue tagging releases using the same convention as you have so far (where the tag is just the version number (excluding the package name) and thereby links directly to the relevant LICENSE).

I also update .gitignore.

Hopefully things are straightforward enough or force you into (what in my opinion would be) the “pit of success”.

### CODE REVIEW

**Comment by binki** [comment]
Ah, the hope is that this could be merged and then you could run the publish script and then upload the nupkg. You can use http://nuget.org/’s [web interface](https://www.nuget.org/packages/upload) or, if you please, [`nuget push`](http://docs.nuget.org/docs/reference/command-line-reference#Push_Command) (after configuring your nuget with [`nuget setapikey`](http://docs.nuget.org/docs/reference/command-line-reference#SetApiKey_Command)). I think it’s preferable that you, @bbarnhill, handle the actual uploading to nuget.org as that would just become part of the normal release process.

File: .gitignore [hunk 1]
```diff
@@ -40,9 +57,12 @@ build/
 *.vssscc
 .builds
 *.pidb
-*.log
+*.svclog
 *.scc
 
+# Chutzpah Test files
```
Comment (bbarnhill) [inline .gitignore]: what is this for?  Is this needed for creating nuspec support?

File: .gitignore [hunk 2]
```diff
@@ -56,12 +76,19 @@ ipch/
...
+# TFS 2012 Local Workspace
+$tf/
+
 # Guidance Automation Toolkit
 *.gpState
 
 # ReSharper is a .NET coding add-in
 _ReSharper*/
 *.[Rr]e[Ss]harper
+*.DotSettings.user
```
Comment (bbarnhill) [inline .gitignore]: there seems to be a lot of inclusions in this .gitignore that dont apply to this project directly.

**Comment by bbarnhill** [comment]
added a couple of comments that if you could clarify before I consider pulling this.  Also Ive never uploaded a nuget package ... may have to get some instructions on that so I can put this as part of the release process.  Thanks for the work and I think its a good idea but just have to wrap up these things before I pull it.

**Comment by bbarnhill** [comment]
I have an account now on the nuget site so after this is fixed and pulled Ill push it up to their site.

File: .gitignore (hunk 2 above)
Comment (binki) [inline .gitignore]: The original `.gitignore` file you had looked like it was pulled from https://github.com/github/gitignore/blob/master/VisualStudio.gitignore I assume a lot of the existing lines in there are not related to your project either. Based on the contents of the .gitignore, you created the git repo through github’s website interface and [chose “VisualStudio” under its “Add .gitignore” option](http://imgur.com/LKV0Dqc). I merely updated it because, without it, git saw [some files created by VS 2015 CTP 5](http://stackoverflow.com/questions/25349322/what-is-the-purpose-of-the-sln-ide-folder#comment41194454_25349322) that shouldn’t be committed. And I was too lazy to make that its own pull request—but it is its own commit. I could separate it out, if you wanted.

The point of the reference VisualStudio .gitignore from github is to help projects that have VS solution/project files produce a .gitignore that works for all of the different development environments that might be used to mess with VS projects. Thus you see resharper and other third party tools—which are .net development-specific—in the github-provided reference .gitignore.

**Comment by binki** [comment]
Hmm, I probably should have put updating `.gitignore` on its own pull request. I’d expect more questions for the changes that *aren’t* to the `.gitignore`… were there any concerns/clarifications you needed for the updates to the `.csproj` and how I chose to write the `.nuspec` file?

**Comment by binki** [comment]
Poke ;-).

**Comment by bbarnhill** [comment]
sorry been extremely busy here the last few days :) Ill get this pulled soon.

File: .gitignore (hunk 1 above)
Comment (bbarnhill) [inline .gitignore]: did you address this?

Comment (binki) [inline .gitignore]: This is just more of [github’s stock VisualStudio.gitignore](https://github.com/bbarnhill/barcodelib/pull/4/files#r23506782). Do you want me to split that out to a separate pull request (it’s already a separate commit…)?

Comment (bbarnhill) [inline .gitignore]: basically anything like nunit and chutzpah javascript test runner inclusions in this .gitignore Id remove for cleanliness of the library in the repo.  They dont apply to the current repo since there are no nunit or chutzpah files included so it more for cleanliness than correctness.

Comment (binki) [inline .gitignore]: I’ll drop my .gitignore edits from the pull…

**Comment by binki** [comment]
OK, dropped irrelvant `.gitignore` updates.

**Comment by bbarnhill** [comment]
Merged ... do I need to do anything to get this nuget package uploaded?  I have an account on nuget I guess I can figure that out after while.

**Comment by bbarnhill** [comment]
published to nuget.org tonight ... I think some of the links you put in the nuspec file are broken so we may have to tweak that a bit.

**Comment by binki** [comment]
Thanks very much! I’m able to add it to my project via VS’s nuget package manager and that seems to work. This will make it much easier to manage in our codebase (we had originally copied the `.cs` files directly into our own project and even edited some of them).

The links in the nuspec very well could be broken. The idea was that they would point at the git tag for that particular release so that the resources from the relevant release could be referenced. Maybe some of the links will fix themselves if you push the tag `1.0.0.20` to github? (The version of the nuget package is 1.0.0.20 but you haven’t pushed a tag for that version to github yet). But trying to make the links version-specific may not be worth the effort, let me know if any particular links need to be un-versioned and I’ll make another pull request for that.

If you want to mess with the `.nuspec` yourself, a technique I find useful is to treat the generated `.nupkg` as a `.zip` (you can just rename it or forcibly open it with 7zFM). This way, you can examine and verify the template substitutions in the `.nuspec` file that is inside the `.nupkg` archive and verify that it looks right before publishing to https://nuget.org/.

**Comment by binki** [comment]
Also, for the `<iconUrl/>` in the published package (whose `.nuspec` looks quite different from what’s committed), it points to https://github.com/bbarnhill/barcodelib/blob/master/BarcodeLibTest/Barcode.ico instead of https://github.com/bbarnhill/barcodelib/blob/master/BarcodeLibTest/Barcode.ico?raw=true ;-).

**Comment by bbarnhill** [comment]
Ive updated the nuspec file with new links.  Ill try to publish them soon.  The links to the release notes and such shouldnt change since I just update the same release notes file every time.
